CRM/Contribute/Form/AbstractEditPayment.php

```diff
@@ -432,12 +432,14 @@ public function assignProcessors() {
       }
       $this->_processors = array();
       foreach ($this->_paymentProcessors as $id => $processor) {
-        $this->_processors[$id] = ts($processor['name']);
-        if (!empty($processor['description'])) {
-          $this->_processors[$id] .= ' : ' . ts($processor['description']);
-        }
-        if ($processor['is_recur']) {
-          $this->_recurPaymentProcessors[$id] = $this->_processors[$id];
+        if ($processor['is_test'] == ($this->_mode == 'test')) {
```

Comment:
It's this line - I'm not sure about wrapping the processor retrieval in a conditional. If it were an assignment it would make more sense to me. But also, I don't think the other forms had the double processors (before this patch) so I guess there are 2 mechanisms in play?